src/H5Dchunk.c

```diff
@@ -1102,49 +1102,28 @@ H5D__chunk_io_init(H5D_io_info_t *io_info, const H5D_type_info_t *type_info, hsi
      * speed up hyperslab calculations by removing the extra checks and/or
      * additions involving the offset and the hyperslab selection -QAK)
      */
-    if ((file_space_normalized = H5S_hyper_normalize_offset(file_space, old_offset)) < 0)
+    if ((file_space_normalized = H5S_hyper_normalize_offset(dinfo->file_space, old_offset)) < 0)
         HGOTO_ERROR(H5E_DATASET, H5E_CANTSET, FAIL, "unable to normalize selection")
 
     /* Decide the number of chunks in each dimension */
     for (u = 0; u < f_ndims; u++)
         /* Keep the size of the chunk dimensions as hsize_t for various routines */
-        fm->chunk_dim[u] = fm->layout->u.chunk.dim[u];
-
-#ifdef H5_HAVE_PARALLEL
-    /* Calculate total chunk in file map*/
-    fm->select_chunk = NULL;
-    if (io_info->using_mpi_vfd) {
-        H5_CHECK_OVERFLOW(fm->layout->u.chunk.nchunks, hsize_t, size_t);
-        if (fm->layout->u.chunk.nchunks)
-            if (NULL == (fm->select_chunk = (H5D_chunk_info_t **)H5MM_calloc(
```

Comment:
Is this mapping no longer used? I couldn't seem to find a replacement in the current code.

Reply:
The replacement for this skiplist is dset_info->layout_io_info.chunk_map->dset_sel_pieces